instrumentation/tomcat/tomcat-jdbc/src/test/java/io/opentelemetry/javaagent/instrumentation/tomcat/jdbc/TomcatJdbcInstrumentationTest.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.tomcat.jdbc;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.mockito.BDDMockito.given;
+
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.db.DbConnectionPoolMetricsAssertions;
+import java.sql.Connection;
+import java.time.Duration;
+import org.apache.tomcat.jdbc.pool.DataSource;
+import org.assertj.core.api.AbstractIterableAssert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class TomcatJdbcInstrumentationTest {
+
+  @RegisterExtension
+  static final AgentInstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  @Mock javax.sql.DataSource dataSourceMock;
+  @Mock Connection connectionMock;
+
+  @Test
+  void shouldReportMetrics() throws Exception {
+    // given
+    given(dataSourceMock.getConnection()).willReturn(connectionMock);
+
+    DataSource tomcatDataSource = new DataSource();
+    tomcatDataSource.setDataSource(dataSourceMock);
+
+    // there shouldn't be any problems if this methods gets called more than once
+    tomcatDataSource.createPool();
+    tomcatDataSource.createPool();
+
+    // when
+    Connection connection = tomcatDataSource.getConnection();
+    connection.close();
+
+    // then
+    await()
+        .atMost(Duration.ofSeconds(20))
+        .untilAsserted(() -> assertConnectionPoolMetrics(tomcatDataSource.getPoolName()));
+
+    // when
+    // this one too shouldn't cause any problems when called more than once
+    tomcatDataSource.close();
+    tomcatDataSource.close();
+    testing.clearData();
+
+    // then
+    await()
+        .atMost(Duration.ofSeconds(20))
+        .untilAsserted(TomcatJdbcInstrumentationTest::assertNoConnectionPoolMetrics);
```

Comment:
is there a need for waiting here vs a single assert?

Reply:
I'll try it.

Reply:
Yup, works without! Thanks.